core/src/main/java/org/jruby/RubyEnumerator.java

```diff
@@ -473,8 +473,8 @@ private SizeFn enumSizeFn(final ThreadContext context) {
         final RubyEnumerator self = this;
         return new SizeFn() {
             @Override
-            public IRubyObject size(IRubyObject[] args) {
-                return self.size(context);
+            public IRubyObject size(ThreadContext context, IRubyObject recv, IRubyObject[] args) {
+                return this.size(context, recv, args);
```

Comment:
this is still doing smt else as before, isn't it? (calling this's method instead of `RubyEnumerator#size`)